documentation/src/test/java/org/hibernate/userguide/mapping/identifier/composite/EmbeddedIdDatabaseGeneratedValueTest.java

```diff
@@ -33,15 +34,16 @@
 	@TestForIssue(jiraKey = "HHH-13096")
 	public void test() {
 		final EventId eventId = doInJPA( this::entityManagerFactory, entityManager -> {
+			// On H2 1.4.199+ CURRENT_TIMESTAMP returns a timestamp with timezone
 			//tag::identifiers-composite-generated-database-example[]
-			Timestamp currentTimestamp = (Timestamp) entityManager
+			OffsetDateTime currentTimestamp = (OffsetDateTime) entityManager
```

Comment:
I'm confused how this is related to "join avoidance"?

Reply:
It's not, I just ran into it since I was working on enabling the user guide tests. I can move this part out if you want though.

Reply:
Its more the general thing.  There are 109 files to look though in this PR.  That's a lot.  And the very first one I look at isn't even related to the thing the PR is trying to do

Reply:
I know, and I am really trying to be more disciplined. Many of the changes, like I tried to outline in the PR description are due to the two changes:
* Introduction of FetchParent#resolveNavigablePath
* ColumnReferenceQualifier#resolveTableReference and ColumnReferenceQualifier#getTableReference now also receive NavigablePath to decide on table reference

If you want, I can extract these other little fixes.